--- a/dependencies.py
+++ b/dependencies.py
@@ -76,7 +76,14 @@ def configure(packageName, versionString=None, eupsProduct=None, eupsProductPath
     state.env.libs = {"main": [], "python": [], "test": []}
     state.env.doxygen = {"tags": [], "includes": []}
     state.env['CPPPATH'] = []
-    state.env['LIBPATH'] = []
+
+    ###################################################
+    # added by stackvana
+    # we always add the conda env to the paths for linking
+    import os as _os
+    state.env['LIBPATH'] = ["%s/lib" % _os.environ['CONDA_PREFIX']]
+    # end of additions by stackvana
+    ###################################################

     # XCPPPATH is a new variable defined by sconsUtils - it's like CPPPATH,
     # but the headers found there aren't treated as dependencies.  This can
