--- a/eupspkg.sh
+++ b/eupspkg.sh
@@ -853,7 +853,11 @@
 		PYDEST="$PREFIX/lib/python"
 		mkdir -p "$PYDEST"
 		# eval is needed to allow $PYSETUP_INSTALL_OPTIONS to include double quotes
-		PYTHONPATH="$PYDEST:$PYTHONPATH" eval python setup.py install $PYSETUP_INSTALL_OPTIONS
+		{
+            PYTHONPATH="$PYDEST:$PYTHONPATH" eval python setup.py install --single-version-externally-managed --record record.txt $PYSETUP_INSTALL_OPTIONS
+        } || {
+            PYTHONPATH="$PYDEST:$PYTHONPATH" eval python setup.py install $PYSETUP_INSTALL_OPTIONS
+        }
 		evil_setuptools_pth_fix "$PYDEST"
 	else
 		# just copy everything, except for the ups directory
